AnkiDroid/src/main/java/com/ichi2/anki/ui/windows/managespace/ManageSpaceFragment.kt

```diff
@@ -0,0 +1,386 @@
+            if (!collectionDirectoryExists()) {
+                emit(Size.FilesAndBytes(emptyList(), 0L))
+            } else {
+                emit(Size.Calculating)
+                emit(block())
+            }
+        } catch (e: CancellationException) {
+            throw e
+        } catch (e: Exception) {
+            emit(Size.Error(e))
+        }
+    }
+}
+
+/**************************************************************************************************
+ ******************************************** Fragment ********************************************
+ **************************************************************************************************/
+
+class ManageSpaceFragment : SettingsFragment() {
+    override val preferenceResource = R.xml.manage_space
+    override val analyticsScreenNameConstant = "manageSpace"
+
+    private val backupLimitsPresenter = BackupLimitsPresenter().also { it.observeLifecycle() }
+
+    private val viewModel: ManageSpaceViewModel by viewModels()
+
+    override fun initSubscreen() {
+        val deleteUnusedMediaPreference = requirePreference<TextWidgetPreference>(R.string.pref_delete_media_key)
+        val deleteBackupsPreference = requirePreference<TextWidgetPreference>(R.string.pref_delete_backups_key)
+        val deleteCollectionPreference = requirePreference<TextWidgetPreference>(R.string.pref_delete_collection_key)
+        val deleteEverythingPreference = requirePreference<TextWidgetPreference>(R.string.pref_delete_everything_key)
+
+        deleteUnusedMediaPreference.launchOnPreferenceClick { onDeleteUnusedMediaClick() }
+        deleteBackupsPreference.launchOnPreferenceClick { onDeleteBackupsClick() }
+        deleteCollectionPreference.launchOnPreferenceClick { onDeleteCollectionClick() }
+        deleteEverythingPreference.launchOnPreferenceClick { onDeleteEverythingClick() }
+
+        adjustDeleteEverythingStringsDependingOnCollectionLocation(deleteEverythingPreference)
+
+        listOf(
+            viewModel.flowOfDeleteUnusedMediaSize to deleteUnusedMediaPreference,
+            viewModel.flowOfDeleteCollectionSize to deleteCollectionPreference,
+            viewModel.flowOfDeleteEverythingSize to deleteEverythingPreference,
+            viewModel.flowOfDeleteBackupsSize to deleteBackupsPreference,
+        ).forEach { (flowOfSize, preference) ->
+            lifecycleScope.launch { flowOfSize.collect { size -> preference.setWidgetTextBy(size) } }
+        }
+    }
+
+    /************************************ Delete unused media *************************************/
+
+    context(CoroutineScope) private suspend fun onDeleteUnusedMediaClick() {
+        val size = viewModel.flowOfDeleteUnusedMediaSize.value
+        if (size is Size.Error && size.exception is Media.MediaCheckRequiredException) {
+            val mediaCheckPromptResult = requireContext().awaitDialog {
+                setMessage("Media check required to find unused media files")
```

Comment:
```suggestion
                setMessage("A media check is required to find unused media files")
```

and needs extraction?

Reply:
Btw I got curious and I asked around on various English language channels and a third of people prefer the variant without “a”, another third with it, and the rest think that either variant is good heh.